### PR TITLE
Read users from file

### PR DESCRIPTION
**Pull request by Mickss**

### CODE REVIEW

File: src/main/java/org/micks/champmaker/auth/AuthService.java [hunk 1]
```diff
@@ -19,4 +22,18 @@ public void createUser(CreateUserRequest createUserRequest) {
...
+        List<User> users = userPersistApi.readFile();
+        String loginRequestUsername = loginRequest.getUsername();
+        String loginRequestPassword = loginRequest.getPassword();
+        Optional<User> first = users.stream()
+                .filter(user -> user.getUsername().equals(loginRequestUsername))
+                .findFirst();
+        User foundUser = first.orElseThrow();
+        String encryptPassword = passwordService.encryptPassword(loginRequestPassword);
+        if (!foundUser.getEncryptedPassword().equals(encryptPassword)) {
+            throw new RuntimeException("Login failed");
```
Comment (tknowak) [inline src/main/java/org/micks/champmaker/auth/AuthService.java:36]: This should be exception more like custom checked exception mapped to 404 status.

File: src/main/java/org/micks/champmaker/auth/AuthService.java [hunk 2]
```diff
@@ -19,4 +22,18 @@ public void createUser(CreateUserRequest createUserRequest) {
...
     }
+
+    public void loginUser(LoginRequest loginRequest) {
+        List<User> users = userPersistApi.readFile();
+        String loginRequestUsername = loginRequest.getUsername();
+        String loginRequestPassword = loginRequest.getPassword();
+        Optional<User> first = users.stream()
+                .filter(user -> user.getUsername().equals(loginRequestUsername))
+                .findFirst();
+        User foundUser = first.orElseThrow();
```
Comment (tknowak) [inline src/main/java/org/micks/champmaker/auth/AuthService.java:33]: This should be exception more like custom checked exception mapped to 404 status.

File: src/main/java/org/micks/champmaker/auth/UserFilePersistService.java [hunk 3]
```diff
@@ -20,4 +24,25 @@ public void storeUser(String username, String encryptedPassword) {
             throw new RuntimeException(e);
         }
     }
+
+    private User convertStr(String userLine) {
```
Comment (tknowak) [inline src/main/java/org/micks/champmaker/auth/UserFilePersistService.java:28]: Let's rename the method name to something more meaningful.

File: src/main/java/org/micks/champmaker/auth/UserPersistApi.java [hunk 4]
```diff
@@ -1,6 +1,9 @@
 package org.micks.champmaker.auth;
 
+import java.util.List;
+
 public interface UserPersistApi {
 
     void storeUser(String username, String encryptedPassword);
+    List<User> readFile();
```
Comment (tknowak) [inline src/main/java/org/micks/champmaker/auth/UserPersistApi.java:8]: Rename the method name - this should be more like `getUsers()`

**Review by tknowak — CHANGES_REQUESTED** [review]

File: src/main/java/org/micks/champmaker/auth/UserPersistApi.java (hunk 4 above)
Comment (Mickss) [reply src/main/java/org/micks/champmaker/auth/UserPersistApi.java:8]: fixed

File: src/main/java/org/micks/champmaker/auth/UserFilePersistService.java (hunk 3 above)
Comment (Mickss) [reply src/main/java/org/micks/champmaker/auth/UserFilePersistService.java:28]: fixed

**Review by tknowak — APPROVED** [review]

**Pull request merged**